service/src/main/java/com/kodemon/service/config/ServiceConfig.java

```diff
@@ -21,6 +30,22 @@
 
     @Bean
     public Mapper dozerMapper() {
-        return new DozerBeanMapper();
+        DozerBeanMapper mapper = new DozerBeanMapper();
+        mapper.addMapping(customMapping());
+
+        return mapper;
+    }
+
+    private BeanMappingBuilder customMapping() {
+        return new BeanMappingBuilder() {
+            @Override
+            protected void configure() {
+                mapping(Pokemon.class, PokemonDTO.class);
+                mapping(Badge.class, BadgeDTO.class);
+                mapping(Trainer.class, UserDTO.class)
```

Comment:
Nope, because no entity has field, that is a Collection of FightDTO or TrainerFight, so everytime you will use mapping, you will say explicitly the two types you want to map. this is done like so, because Trainer (UserDTO) have both collections of pokemon(pokemonDTO) and badges(BadgeDTO). If trainer and userDTO had field that is a collection of the same type, tis mapping is not necessary.